### PR TITLE
Set Api Based Authentication Flag

### PR DESCRIPTION
**Pull request by DMHP**
Set Api Based Authentication Flag

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/core/ServerAuthenticatorManagementService.java [hunk 1]
```diff
@@ -511,6 +512,9 @@ private Authenticator addLocalAuthenticator(LocalAuthenticatorConfig config) {
         authenticator.setName(config.getName());
         authenticator.setDisplayName(config.getDisplayName());
         authenticator.setIsEnabled(config.isEnabled());
+        if (config.getProperties().length > 0 && IS_API_BASED_SUPPORTED.equals(config.getProperties()[0].getName())) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/core/ServerAuthenticatorManagementService.java:515]: shall we add a null check here is there is any possibility of config.getProperties() being null

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/gen/java/org/wso2/carbon/identity/api/server/configs/v1/model/InboundAuthSAML2Config.java [hunk 2]
```diff
@@ -36,7 +37,7 @@ public class InboundAuthSAML2Config  {
   
     private List<String> destinationURLs = null;
 
-    private Integer metadataValidityPeriod;
+    private BigDecimal metadataValidityPeriod;
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/gen/java/org/wso2/carbon/identity/api/server/configs/v1/model/InboundAuthSAML2Config.java:40]: Let remove this file from the commit. Seems like a previous editor of the api definition hasn't updated the gen code.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java [hunk 3]
```diff
@@ -605,6 +607,11 @@ private List<AuthenticatorListItem> buildAuthenticatorListResponse(
                 authenticatorListItem.setName(config.getName());
                 authenticatorListItem.setDisplayName(config.getDisplayName());
                 authenticatorListItem.setIsEnabled(config.isEnabled());
+                if (config.getProperties().length > 0 && IS_API_BASED_SUPPORTED.equals
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:610]: Lets do null check here if there is any possibility of config.getProperties() getting null

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/resources/configs.yaml [hunk 4]
```diff
@@ -1040,6 +1040,10 @@ components:
           type: boolean
           default: true
           example: true
+        isAPIBasedAuthentication:
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/resources/configs.yaml:1043]: Shall we change this to isAPIBasedAuthenticationSupported

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/gen/java/org/wso2/carbon/identity/api/server/configs/v1/model/InboundAuthSAML2Config.java (hunk 2 above)
Comment (DMHP) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/gen/java/org/wso2/carbon/identity/api/server/configs/v1/model/InboundAuthSAML2Config.java:40]: @ThaminduDilshan has added this code. He will test the fix after merging the PR.

**Pull request merged**